feat: 마이페이지 회원정보 보안

choidaehwan opened the pull request:
# 반영 브랜치
dev
# 변경 사항
    - 닉네임변경 기능 구현
    - 비밀번호 변경 기능 구현
    - 회원탈퇴 기능 구현
    - 유저가 소속된팀 보여게 하는 기능 구현
    - 월마다 끝날 바뀌게 하는 기능 구현

# 확인 방법 (스크린샷 포함)

choidaehwan committed:
w

'

choidaehwan committed:
feat: 마이페이지 회원정보 보안

- 닉네임변경 기능 구현
- 비밀번호 변경 기능 구현
- 회원탈퇴 기능 구현
- 유저가 소속된팀 보여게 하는 기능 구현
- 월마다 끝날 바뀌게 하는 기능 구현

K-MinJae commented on application.py:
```diff
@@ -188,17 +188,81 @@ def nickname_check():
+def withdrawal():
+    user_id = request.user['_id']
+
+    db.user.delete_one({'_id': user_id})
```
user말고 다른 collection에서도 탈퇴한 유저의 정보가 있는지 확인하고 있다면 삭제하는게 좋겠습니다

Dae-Hwan replied:
넵

K-MinJae commented on application.py:
```diff
@@ -188,17 +188,81 @@ def nickname_check():
+    return jsonify({'msg': 'SUCCESS'})
+
+
+# 비빌번호 변경 새로운 비밀번호 유효성검사
```
비밀번호 유효성 검사하는 부분이 아닌 것 같아 수정하면 좋겠습니다

Dae-Hwan replied:
유효성 검사하는 부분 맞습니다

Dae-Hwan replied:
수정했습니다...

K-MinJae commented on templates/my_page.html:
```diff
@@ -200,7 +201,32 @@ <h3 class="sign-out-title">회원탈퇴</h3>
+      <div class="modal-body">
+          <div class="input-box">
+              닉네임: <input id="nickname" type="text" width="100px">
+              <p id="can-using">사용해도 되는 닉네임 입니다.</p>
```
현재 사용하고 있는 닉네임으로 다시 바꾸려고 할 경우 알럿창이 뜨면 좋겠습니다

choidaehwan committed:
feat: 코드리뷰 반영

- progressbar 소숫점 없애기
- 비밀번호, 닉네임 이전과 같은것을 쓰지 못하게 막음
- 회원정보 삭제할때 DB의 다른 collection의 연관되어있는 객체들도 함께 삭제

choidaehwan committed:
feat: 뜻이 안맞는 주석 수정
